### PR TITLE
Data Science Team- parking slot occupancy detection - 80% completed

### PR DESCRIPTION
**Pull request by sgulluoglu**
•	Conducted further data processing to ensure accuracy and readiness for analysis including correlation analysis and encoding categorical variables. 
•	Reviewed the pre-processed dataset with visualization tools. Visualize trends and distributions to understand the parking behavior. Used Heat maps, Histogram charts and distributions, Bar Plots, Line Graphs.
•	Calculated rolling averages of monthly based occupied parking bays.  
•	Resampling the data to an hourly frequency to understand the tendency of parking status. 
•	Developed a predictive model using TensorFlow to forecast parking availability, leveraging time series data and deep learning techniques. Training of  the dataset for  predicting parking tendency for 2025, forecast the future value over a period of time.
•	Plotting Forecast: Included a plot for both historical data and forecasted occupancy to visualize the prediction.
•	Comparison- estimated parking slots needs for 2025 based on previous years analysis.
•	Researched the impack of data drift in parking analysis: frequently changed dataset may cause the predictive model behave wrongly. It is essential to ensure this change is not affecting the accuracy and realibility overtime. 
•	Prediction Analysis Short-Term Prediction (Next Minute to Hour): Use data intervals of 5 to 15 minutes. Medium-Term Prediction (Next Day to Week): Aggregate the data to an hourly interval. Long-Term Prediction (Next Month to Year): Aggregate the data to a daily or weekly interval.
Built LSTM(Long short term memory) model for time series forecasting
Built Single Dense model 
Built ARIMA (AutoRegressive Integrated Moving Average) 
Built Prophet 
Built Support Vector Regression (SVR)
•	RMSE(Root Mean Square Error) calculations for analyzing model accuracy 
•	Comparison analysis between the model results - Arima-SVR-Prophet 
•	Data Drifting – analyze the accuracy of the models because of the frequently changing dataset. 
Quantitative Measuring for Data Drift: 3 ways to measure, Kolmogorov-Smirnov Test is applied.
•	Visualization
Histograms and KDE Plots: Visualize and compare the distributions of the datasets. Kernel Density Estimate(KDE):visualizing the Probability Density of a continuous variable - KDE plots to compare distributions.
 Correlation Matrix Heatmaps: Compare the correlation patterns to detect any structural changes in the data. 
Created Geodataframe to plot available parking spots. 
•	Resampled data to a minute frequency, hour frequency and monthly frequency.
•	Data Drift Monitoring with the following measuring metrics(Earth Mover's Distance (EMD) / Jensen-Shannon Divergence (JSD) & Kolmogorov-Smirnov Statistic – to see the performance and behaviour of the model on prediction for a frequently change dataset.
•	Applied data drift monitoring techniques on each prediction algorithms.
•	Completed the interpretation of data drift monitoring.

### CODE REVIEW

**Review by dTitanium — CHANGES_REQUESTED** [review]
put all libraries together in single code chunk
remove unnecessary dataframe outputs
remove /short output lengths 
  ex: 1. You can only print dataframe head
        2. You can hide model training steps in your output
 Simply xplain what you re doing in codes

**Review by Sachithak95 — CHANGES_REQUESTED** [review]
Hi Sabri,

As a junior, you've done good work on the use case. However, as Dinuk mentioned, I also noticed some major errors. The formatting is not done properly, and you've displayed the entire dataset in your use case, which is unnecessary. Please just print the first five rows of the DataFrame when showcasing the data.

Additionally, since your use case is incomplete, you don't need to generate the .json and .html files yet.

Also, as discussed, please make sure to include all the libraries and packages at the beginning of the use case.

Please make these changes so that we can approve your pull request.

Thank you!
Sachitha

**Comment by sgulluoglu** [comment]
Dear reviewers,
Thank you for your comments. 
For answering the requested changes, I did my best effort in this limited time. 
I have almost completed the formatting, printed only first five rows of the Data frame when running the code and I deleted the .json and html files from the directory. I will continue to work on this use case next trimester, so I will review the formatting regularly, will add additional explanations and interpretations on each code blocks, will update the required libraries or/and packages. Also my future plans about the use case can be seen in the relevant headings 14 and 15.

**Review by Sachithak95 — APPROVED** [review]

**Review by dTitanium — APPROVED** [review]
Good effort! I hope you can finish this use case in the next trimester!

**Review by hiruboy96 — APPROVED** [review]
Hi @sgulluoglu,

As a junior, you’ve done an impressive job, and the use case looks good, even though it’s not fully completed yet. The visualizations are clear and concise, and I appreciate the creativity. Additionally, it’s great that the use case is accessible via nbviewer, showing the effort you’ve put into your first use case this trimester.

Going forward to improve your use case:

Make sure you're familiar with the color codes for use cases by referring to the guide documents.
Group all library installation codes into a single chunk, mark it as optional, and place it at the beginning of the use case.
Consolidate all the library imports into one block at the start.
To keep the notebook cleaner, use warnings.filterwarnings('ignore') to hide unnecessary warnings.
Include brief explanations and insights for each of your visualizations.
Make sure you use the API to import your datasets next trimester.
Good luck for the next trimester!

Best regards,
Venuka
